internal/engine/experiment/tor/tor.go

```diff
@@ -251,7 +245,7 @@ func (m *Measurer) measureTargets(
 type resultsCollector struct {
 	callbacks       model.ExperimentCallbacks
 	completed       *atomicx.Int64
-	flexibleConnect func(context.Context, keytarget) (oonitemplates.Results, error)
+	flexibleConnect func(context.Context, keytarget) (*measurex.ArchivalMeasurement, *string)
```

Comment:
This function is the "engine" implementing the measurement. Interestingly, I only needed to change its return value and implementation to adapt the `tor` test (is this a good sign?).